File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/property/create/PropertyTypeCreation.js
define([ "stjs" ], function(stjs) {
	var PropertyTypeCreation = function() {
	};
	stjs.extend(PropertyTypeCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.property.create.PropertyTypeCreation';
		constructor.serialVersionUID = 1;
		prototype.code = null;
		prototype.label = null;
		prototype.description = null;
		prototype.managedInternally = null;
		prototype.dataType = null;
		prototype.vocabularyId = null;
		prototype.materialTypeId = null;
		prototype.sampleTypeId = null;
		prototype.schema = null;
		prototype.transformation = null;
		prototype.metaData = null;

		prototype.getCode = function() {
			return this.code;
		};
		prototype.setCode = function(code) {
			this.code = code;
		};
		prototype.getLabel = function() {
			return this.label;
		};
		prototype.setLabel = function(label) {
			this.label = label;
		};
		prototype.getDescription = function() {
			return this.description;
		};
		prototype.setDescription = function(description) {
			this.description = description;
		};
		prototype.isManagedInternally = function() {
			return this.managedInternally;
		};
		prototype.setManagedInternally = function(managedInternally) {
			this.managedInternally = managedInternally;
		};
		prototype.getDataType = function() {
			return this.dataType;
		};
		prototype.setDataType = function(dataType) {
			this.dataType = dataType;
		};
		prototype.getVocabularyId = function() {
			return this.vocabularyId;
		};
		prototype.setVocabularyId = function(vocabularyId) {
			this.vocabularyId = vocabularyId;
		};
		prototype.getMaterialTypeId = function() {
			return this.materialTypeId;
		};
		prototype.setMaterialTypeId = function(materialTypeId) {
			this.materialTypeId = materialTypeId;
		};
		prototype.getSampleTypeId = function() {
			return this.sampleTypeId;
		};
		prototype.setSampleTypeId = function(sampleTypeId) {
			this.sampleTypeId = sampleTypeId;
		};
		prototype.getSchema = function() {
			return this.schema;
		};
		prototype.setSchema = function(schema) {
			this.schema = schema;
		};
		prototype.getTransformation = function() {
			return this.transformation;
		};
		prototype.setTransformation = function(transformation) {
			this.transformation = transformation;
		};
		prototype.getMetaData = function() {
			return this.metaData;
		};
		prototype.setMetaData = function(metaData) {
			this.metaData = metaData;
		};
	}, {
		dataType : "DataType",
		vocabularyId : "IVocabularyId",
		materialTypeId : "IEntityTypeId",
		sampleTypeId : "IEntityTypeId"
	});
	return PropertyTypeCreation;
})

File: api-openbis-java/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/property/create/PropertyTypeCreation.java
/*
 * Copyright ETH 2018 - 2023 Zürich, Scientific IT Services
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.asapi.v3.dto.property.create;

import java.util.Map;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.IObjectCreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.entitytype.id.IEntityTypeId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.DataType;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.vocabulary.id.IVocabularyId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author Franz-Josef Elmer
 */
@JsonObject("as.dto.property.create.PropertyTypeCreation")
public class PropertyTypeCreation implements ICreation, IObjectCreation
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private String code;

    @JsonProperty
    private String label;

    @JsonProperty
    private String description;

    @JsonProperty
    private boolean managedInternally;

    @JsonProperty
    private DataType dataType;

    @JsonProperty
    private IVocabularyId vocabularyId;

    @JsonProperty
    private IEntityTypeId materialTypeId;

    @JsonProperty
    private IEntityTypeId sampleTypeId;

    @JsonProperty
    private String schema;

    @JsonProperty
    private String transformation;

    @JsonProperty
    private Map<String, String> metaData;

    @JsonProperty
    private Boolean multiValue;

    @JsonIgnore
    public String getCode()
    {
        return code;
    }

    public void setCode(String code)
    {
        this.code = code;
    }

    @JsonIgnore
    public String getLabel()
    {
        return label;
    }

    public void setLabel(String label)
    {
        this.label = label;
    }

    @JsonIgnore
    public String getDescription()
    {
        return description;
    }

    public void setDescription(String description)
    {
        this.description = description;
    }

    @JsonIgnore
    public boolean isManagedInternally()
    {
        return managedInternally;
    }

    public void setManagedInternally(boolean managedInternally)
    {
        this.managedInternally = managedInternally;
    }

    /**
     * @deprecated use {@link #isManagedInternally()}
     */
    @JsonIgnore
    @Deprecated()
    public boolean isInternalNameSpace()
    {
        return isManagedInternally();
    }

    /**
     * @deprecated use {@link #setManagedInternally(boolean)}
     */
    @Deprecated()
    public void setInternalNameSpace(boolean internalNameSpace)
    {
        setManagedInternally(internalNameSpace);
    }

    @JsonIgnore
    public DataType getDataType()
    {
        return dataType;
    }

    public void setDataType(DataType dataType)
    {
        this.dataType = dataType;
    }

    @JsonIgnore
    public IVocabularyId getVocabularyId()
    {
        return vocabularyId;
    }

    public void setVocabularyId(IVocabularyId vocabularyId)
    {
        this.vocabularyId = vocabularyId;
    }

    @JsonIgnore
    public IEntityTypeId getMaterialTypeId()
    {
        return materialTypeId;
    }

    public void setMaterialTypeId(IEntityTypeId materialTypeId)
    {
        this.materialTypeId = materialTypeId;
    }

    @JsonIgnore
    public IEntityTypeId getSampleTypeId()
    {
        return sampleTypeId;
    }

    public void setSampleTypeId(IEntityTypeId sampleTypeId)
    {
        this.sampleTypeId = sampleTypeId;
    }

    @JsonIgnore
    public String getSchema()
    {
        return schema;
    }

    public void setSchema(String schema)
    {
        this.schema = schema;
    }

    @JsonIgnore
    public String getTransformation()
    {
        return transformation;
    }

    public void setTransformation(String transformation)
    {
        this.transformation = transformation;
    }

    @JsonIgnore
    public Map<String, String> getMetaData()
    {
        return metaData;
    }

    public void setMetaData(Map<String, String> metaData)
    {
        this.metaData = metaData;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public Boolean isMultiValue()
    {
        return multiValue;
    }

    // Method automatically generated with DtoGenerator
    public void setMultiValue(Boolean multiValue)
    {
        this.multiValue = multiValue;
    }

    @Override
    public String toString()
    {
        return new ObjectToString(this).append("code", code).toString();
    }

}
